eng/Versions.props

```diff
@@ -4,7 +4,6 @@
   <PropertyGroup>
     <!-- Bumping the Roslyn version used in order to ingest the new runtime source generators -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
-    <MicrosoftNetCompilersToolsetVersion>4.4.0-3.22452.8</MicrosoftNetCompilersToolsetVersion>
```

Comment:
@joperezr Can you please comment if there are still any reasons for keeping the pinned version of toolset introduced by this change? https://github.com/dotnet/templating/pull/5234/commits/5524dedd8f5f0553d5d05c0f48580bf02067b013

If it needs to be pinned - should the version be actually updated? Or should we remove the `UsingToolMicrosoftNetCompilers`? Those seem to lead to conflict in dev build scenario

Reply:
So sorry for the delay, I completely missed earlier pings here. Yes, should be totally fine to remove the pinning of a specific version of the compiler package. At the time, the change was _pinning up_ the dependency since we needed a specific change that wasn't pushed by Arcade yet. Now, Arcade is pushing an even newer version, so this line is effectively _pinning down_ the version, which we of course don't want to do.

Reply:
So removing doesn't break the build, so let it be removed to simplify build